src/EditorFeatures/CSharpTest/Completion/CompletionProviders/DeclarationNameCompletionProviderTests.cs

```diff
@@ -2282,7 +2287,75 @@ public void Method(IEnumerable<int>? $$)
   }
 }
 ";
-            await VerifyItemExistsAsync(markup, "vs");
+            await VerifyItemExistsAsync(markup, "ints");
+        }
+
+        [WorkItem(36364, "https://github.com/dotnet/roslyn/issues/36364")]
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task EnumerableParameterOfUnmanagedType()
+        {
+            var markup = @"
+using System.Collections.Generic;
+
+public class Class1
+{
+  public void Method(IEnumerable<int> $$)
+  {
+  }
+}
+";
+            await VerifyItemExistsAsync(markup, "ints");
+        }
+
+        [WorkItem(36364, "https://github.com/dotnet/roslyn/issues/36364")]
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task EnumerableParameterOfObject()
+        {
+            var markup = @"
+using System.Collections.Generic;
+
+public class Class1
+{
+  public void Method(IEnumerable<object> $$)
+  {
+  }
+}
+";
+            await VerifyItemExistsAsync(markup, "objects");
+        }
+
+        [WorkItem(36364, "https://github.com/dotnet/roslyn/issues/36364")]
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task EnumerableParameterOfString()
+        {
+            var markup = @"
+using System.Collections.Generic;
+
+public class Class1
+{
+  public void Method(IEnumerable<string> $$)
+  {
+  }
+}
+";
+            await VerifyItemExistsAsync(markup, "strings");
+        }
+
+        [WorkItem(36364, "https://github.com/dotnet/roslyn/issues/36364")]
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task EnumerableGenericParameter()
+        {
+            var markup = @"
+using System.Collections.Generic;
+
+public class Class1
+{
+  public void Method<T>(IEnumerable<T> $$)
+  {
+  }
+}
+";
+            await VerifyItemExistsAsync(markup, "generics");
```

Comment:
this doesn't seem ideal.  I feel like for a generic T, we should just use `value` (and thus make this `values`).  If it TName (like `TKey`) then we should pull off the suffix and generate `key/keys`).

Reply:
+1 I don't like "generic" or "generics" as parameter names

Reply:
@CyrusNajmabadi Great suggestion!